auth-tokens-filter/src/main/java/com/palantir/tokens/auth/http/BasicAuthToBearerTokenFilter.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tokens.auth.http;
+
+import com.google.common.io.BaseEncoding;
+import com.google.common.net.HttpHeaders;
+import com.palantir.tokens.auth.AuthHeader;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Objects;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A {@link Filter} that replaces basic auth with a bearer token. It assumes the bearer token is in the password field
+ * of the basic auth header.
+ */
+public class BasicAuthToBearerTokenFilter implements Filter {
+    private static final String BASIC_AUTH_STR = "Basic";
+    private static final Logger log = LoggerFactory.getLogger(BasicAuthToBearerTokenFilter.class);
+    private static final BaseEncoding BASE_64_ENCODING = BaseEncoding.base64Url();
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+    }
+
+    @Override
+    public final void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+            throws IOException, ServletException {
+        ServletRequest updatedRequest;
+        try {
+            updatedRequest = getRequestWithToken(request);
+        } catch (NullRawAuthHeaderException e) {
+            updatedRequest = request;
+        } catch (AuthHeaderNotBasicAuthException e) {
+            log.warn("AuthHeaderNotBasicAuthException", e);
+            updatedRequest = request;
+        }
+        chain.doFilter(updatedRequest, response);
+    }
+
+    @Override
+    public void destroy() {
+    }
+
+    private ServletRequest getRequestWithToken(ServletRequest request)
+            throws AuthHeaderNotBasicAuthException, NullRawAuthHeaderException {
+        if (request instanceof HttpServletRequest) {
+            final HttpServletRequest httpRequest = (HttpServletRequest) request;
```

Comment:
no need to specify local variables as final
http://stackoverflow.com/questions/18225572/use-of-final-local-variables-in-java